src/DotLottie.tsx

```diff
@@ -0,0 +1,204 @@
+import {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useRef,
+  type MutableRefObject,
+} from 'react';
+import {
+  requireNativeComponent,
+  UIManager,
+  Platform,
+  type ViewStyle,
+  findNodeHandle,
+} from 'react-native';
+import type { Mode } from './constants';
+import { parseSource } from './utils';
+
+const LINKING_ERROR =
+  `The package 'dotlottie-react-native' doesn't seem to be linked. Make sure: \n\n` +
+  Platform.select({ ios: "- You have run 'pod install'\n", default: '' }) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo Go\n';
+
+export type Dotlottie = {
+  play: () => void;
+  pause: () => void;
+  stop: () => void;
+  setLoop: (value: boolean) => void;
+  setSpeed: (value: number) => void;
+  setPlayMode: (mode: Mode) => void;
+  setFrame: (frame: number) => void;
+  freeze: () => void;
+  unfreeze: () => void;
+};
+
+interface DotlottieReactNativeProps {
+  source: string | { uri: string };
+  loop?: boolean;
+  autoplay?: boolean;
+  speed?: number;
+  style: ViewStyle;
+  ref?: MutableRefObject<any>;
+  onLoad?: () => void;
+  onComplete?: () => void;
+  onLoadError?: () => void;
+  onPlay?: () => void;
+  onLoop?: () => void;
+  onDestroy?: () => void;
+  onUnFreeze?: () => void;
+  onFreeze?: () => void;
+  onPause?: () => void;
+  onFrame?: () => void;
+  onStop?: () => void;
+  onRender?: () => void;
+}
+
+const COMMAND_PLAY = 'play';
+const COMMAND_PAUSE = 'pause';
+const COMMAND_STOP = 'stop';
+const COMMAND_SET_LOOP = 'setLoop';
+const COMMAND_SET_SPEED = 'setSpeed';
+const COMMAND_SET_PLAY_MODE = 'setPlayMode';
+const COMMAND_SET_FRAME = 'setFrame';
+const COMMAND_FREEZE = 'freeze';
+const COMMAND_UNFREEZE = 'unfreeze';
+
+const ComponentName = 'DotlottieReactNativeView';
+
+const NativeViewManager = UIManager.getViewManagerConfig(ComponentName);
+
+const DotlottieReactNativeView =
+  UIManager.getViewManagerConfig(ComponentName) != null
+    ? requireNativeComponent<DotlottieReactNativeProps>(ComponentName)
+    : () => {
+        throw new Error(LINKING_ERROR);
+      };
+
+export const DotLottie = forwardRef(
+  ({ source, ...props }: DotlottieReactNativeProps, ref) => {
```

Comment:
If any of the props, like `source` or `loop`, are updated, will those changes be reflected in the native environment? I'm assuming they won't, since we need to call `setLoop`, `setSource` on the dotLottie instance, right ?

Reply:
yes it does immediately